src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -690,9 +769,18 @@ protected virtual Task<TResult> InvokeCoreAsync<TResult>(int? id, string targetN
                     }
                     else
                     {
-                        requestMethodToClrMethodMap.Add(requestName, method.Name);
+                        requestMethodToClrMethodNameMap.Add(requestName, method.Name);
+                    }
+
+                    // Skip this method if its signature matches one from a derived type we have already scanned.
```

Comment:
Do we need a unit test to cover that scenario?

Reply:
We do. Code coverage shows that a great many tests cover this code.